ui/src/components/StatusSwitch.jsx

```diff
@@ -1,51 +1,53 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import FormControlLabel from "@material-ui/core/FormControlLabel";
 import Switch from "@material-ui/core/Switch";
 import axios from "axios";
 
-class StatusSwitch extends React.Component {
-  constructor(props) {
-    super(props);
+const StatusSwitch = () => {
+  const [isRunning, setIsRunning] = useState(true);
 
-    this.state = {
-      checked: true,
-      status: "Running"
-    };
+  useEffect(() => {
+    const stateUrl = new URL(
+      "/state",
+      window.config.env.REACT_APP_NOPAS_ENDPOINT
+    );
 
-    this.handleChange = this.handleChange.bind(this);
-  }
+    axios.get(stateUrl)
+    .then(rsp => setIsRunning(rsp.data))
+    .catch(function (error) {
+      console.log(error);
+    })
+  }, []);
 
-  handleChange(event) {
+  const handleChange = (event) => {
     if (event.target.checked) {
       const resumeUrl = new URL(
-        "/resume",
+        "/state/resume",
         window.config.env.REACT_APP_NOPAS_ENDPOINT
       );
-      axios.put(resumeUrl);
+      axios.put(resumeUrl)
+      .then(rsp => setIsRunning(true))
+      .catch(err => {console.log(err)})
+
     } else {
       const pauseUrl = new URL(
-        "/pause",
+        "/state/pause",
         window.config.env.REACT_APP_NOPAS_ENDPOINT
       );
-      axios.put(pauseUrl);
+      axios.put(pauseUrl)
+      .then(rsp => setIsRunning(false))
+      .catch(err => {console.log(err)})
     }
```

Comment:
Alternative style is to use `async/await`, then wrap the whole thing with `catch` block.

This changes its return value to be `Promise<void>` I think, but based on this: https://stackoverflow.com/a/44233381, `Promise<void>` -> `void`.

The `onChange` event handler for `Switch` that doesn't need the return value: https://material-ui.com/api/switch/, since it expects `void` as return type.